machine-learning/04-Model Evaluation.ipynb

```diff
@@ -4,25 +4,478 @@
+          },
+          "execution_count": 1,
+          "metadata": {}
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Initialize MLContext\n",
+        "\n",
+        "All ML.NET operations start in the [MLContext](https://docs.microsoft.com/dotnet/api/microsoft.ml.mlcontext) class. Initializing mlContext creates a new ML.NET environment that can be shared across the model creation workflow objects. It's similar, conceptually, to DBContext in Entity Framework."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        }
+      },
+      "source": [
+        "var mlContext = new MLContext();"
+      ],
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Split the data into train and test sets\n",
+        "\n",
+        "The train set is what you'll use to learn the patterns of your data. The test set is the dataset used to evaluate the performance of your model using evaluation metrics for the regression task. \n",
+        "\n",
+        "In this case, 80% of the data is used for training and 20% is used for test/evaluation as defined by the `testFraction` parameter."

```

Comment:
I would also include what a validation set is used for

Reply:
resolved